From df6792e3bcc80b4339df697a2bbe62adf1cbe8e4 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Wed, 17 May 2017 15:14:05 -0400
Subject: [PATCH 2/2] net: Remove unused network options

net/filter seems to be needed for general packet receive.

net_init_dump my be removed with the removal of dump.o
---
 net/Makefile.objs | 6 +++---
 net/net.c         | 2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/net/Makefile.objs b/net/Makefile.objs
index b7c22fd..4be8117 100644
--- a/net/Makefile.objs
+++ b/net/Makefile.objs
@@ -1,6 +1,6 @@
 common-obj-y = net.o queue.o checksum.o util.o hub.o
 common-obj-y += socket.o
-common-obj-y += dump.o
+common-obj-n += dump.o
 common-obj-y += eth.o
 common-obj-$(CONFIG_L2TPV3) += l2tpv3.o
 common-obj-$(CONFIG_POSIX) += tap.o vhost-user.o
@@ -14,5 +14,5 @@ common-obj-$(CONFIG_SLIRP) += slirp.o
 common-obj-$(CONFIG_VDE) += vde.o
 common-obj-$(CONFIG_NETMAP) += netmap.o
 common-obj-y += filter.o
-common-obj-y += filter-buffer.o
-common-obj-y += filter-mirror.o
+common-obj-n += filter-buffer.o
+common-obj-n += filter-mirror.o
diff --git a/net/net.c b/net/net.c
index 15a3920..cb6f74a 100644
--- a/net/net.c
+++ b/net/net.c
@@ -964,7 +964,9 @@ static int (* const net_client_init_fun[NET_CLIENT_OPTIONS_KIND__MAX])(
 #ifdef CONFIG_NETMAP
         [NET_CLIENT_OPTIONS_KIND_NETMAP]    = net_init_netmap,
 #endif
+#if 0
         [NET_CLIENT_OPTIONS_KIND_DUMP]      = net_init_dump,
+#endif
 #ifdef CONFIG_NET_BRIDGE
         [NET_CLIENT_OPTIONS_KIND_BRIDGE]    = net_init_bridge,
 #endif
-- 
2.9.4

